### PR TITLE
Add Pin definitions for Adafruit Feather ESP32-C6

### PR DESCRIPTION
**Pull request by tyeth**
Adds pin defs for C6, although possibly wrong analog mcu Ref Voltage.

> Any idea the analog ref voltage for the c6? I've seen the c3 qtpy using 2.6v but struggling to find that in c3 tech ref datasheet, and therefore matching info for C6
> 
> C3 which should mention 2.6v - https://www.espressif.com/sites/default/files/documentation/esp32-c3_technical_reference_manual_en.pdf#sensor see 34.2.3.2
> 
> Whereas C6 does mention vref of 1.1v for adc and then attenuated https://www.espressif.com/sites/default/files/documentation/esp32-c6_technical_reference_manual_en.pdf#sensor see 39.2.3.2
> Wasn't sure if 1.1v was correct. Either way I'll test the adc once I get it built so no stress
>

### CODE REVIEW

**Comment by tyeth** [comment]
@brentru if you wouldn't mind casting an eye over this

**Comment by tyeth** [comment]
I'm also fully aware that the documentation link is incorrectly pointing at the QTPY C3, as it's the only learn guide related to ESP32-Cx chips at the moment (feather c6 learn guide doesn't yet exist), and the process is identical so presumed that's the best we can link to for now.

File: boards/feather-esp32c6/definition.json [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        "name": "D16",
+        "displayName": "D16 (UART TX)",
+        "dataType": "bool",
+        "hasPWM": true,
+        "hasServo": true,
+        "uartTx": true
+      },
+      {
+        "name": "D9",
+        "displayName": "Button (IO9/NeoPixel)",
```
Comment (brentru) [inline boards/feather-esp32c6/definition.json:106]: @tyeth  If we're going to use this as a NeoPixel, remove "Button" and call it NeoPixel

**Review by brentru — CHANGES_REQUESTED** [review]

File: boards/feather-esp32c6/definition.json (hunk 1 above)
Comment (tyeth) [reply boards/feather-esp32c6/definition.json]: Yep have done. I was initially going to test both on staging, but at a cafe right now, so just removed button as suggested.

**Pull request merged**